code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm

```diff
@@ -38,9 +39,143 @@
 	..()
 	src << "<span class='notice'>You can kill and eat other drones to increase your health!</span>" //Inform the evil lil guy
 
+/*
+//Syndrones shouldn't be blocked like regular drones.
+/mob/living/simple_animal/drone/syndrone/ClickOn(atom/A, params)
+	...(A, params)
+*/
 
 /obj/item/drone_shell/syndrone
 	name = "syndrone shell"
 	desc = "A shell of a syndrone, a modified maintenance drone designed to infiltrate and annihilate."
 	icon_state = "syndrone_item"
-	drone_type = /mob/living/simple_animal/drone/syndrone
\ No newline at end of file
+	drone_type = /mob/living/simple_animal/drone/syndrone
+
+//Safedrones - Excuse the awful ClickOn proc
+/mob/living/simple_animal/drone/safedrone
+
+/mob/living/simple_animal/drone/ClickOn(atom/A, params)
+	if(world.time <= next_click)
+		return
+	next_click = world.time + 1
+	if(client.buildmode)
+		build_click(src, client.buildmode, params, A)
+		return
+
+	var/list/modifiers = params2list(params)
+	if(modifiers["shift"] && modifiers["ctrl"])
+		CtrlShiftClickOn(A)
+		return
+	if(modifiers["middle"])
+		MiddleClickOn(A)
+		return
+	if(modifiers["shift"])
+		ShiftClickOn(A)
+		return
+	if(modifiers["alt"]) // alt and alt-gr (rightalt)
+		AltClickOn(A)
+		return
+	if(modifiers["ctrl"])
+		if(ismob(A))
+			if(!isdrone(A))
+				src << "<span class='danger'>Your laws prevent you from doing this!</span>"
+				return
+		CtrlClickOn(A)
+		return
+
+	if(stat || paralysis || stunned || weakened)
+		return
+
+	face_atom(A)
+
+	if(next_move > world.time) // in the year 2000...
+		return
+
+	if(restrained())
+		changeNext_move(CLICK_CD_HANDCUFFED)   //Doing shit in cuffs shall be vey slow
```

Comment:
I don't think drones have ever had onclick overriden and I wasn't too sure about it. I thought to err on the side of caution incase something in drones or simple_animals make use of restrained()